dicom_validator/spec_reader/part3_reader.py

```diff
@@ -19,6 +23,25 @@
 )
 
 
+# Some conditions from the spec are hard to parse with a few general rules.
+# Instead of implementing a full parser for these cases, we define them here.
+SPECIAL_CASES = {
```

Comment:
I like the idea of special cases - I have also been thinking about this, and this is an easy way to do this.
I still think I have to fix the parser so it at least ignores the part before "that is", and correctly recognizes the "or" condition (currently it mingles together the or'ed values and the next condition), but this can be done later.

Reply:
Yeah, my first thought was to edit the parser too. But even if you completely omit the part `Graphic Data (0070,0022) is "closed", that is ` and you're left with 

```
Required if Graphic Type (0070,0023) is CIRCLE or ELLIPSE, or Graphic Type (0070,0023) is POLYLINE or INTERPOLATED and the first data point is the same as the last data point.
```

the current parser won't parse it correctly. I've realized that changing the parser logic to address this issue (multiple "or" conditions with "and" condition and to correctly put them together in the intended precedence) can very easily break the rest of the conditions that now works perfectly.

Hence the solution with special cases that can occur from time to time. The obvious downside is that these have to be checked manually with every new spec version. 😕

Reply:
Yes, you are right. I'll nevertheless see if I can handle these conditions. In principle, I have been thinking to rework the whole condition parser to be more like a real parser - e.g., build some kind of syntax tree first - but I'm not sure if I will ever really do this :/
Also I'm going to fix this exception that may happen in such cases, though this is a minor change (just catch the exception).
And I'll probably use that special case handling for other cases that I can't handle right now :)